common/djangoapps/request_cache/tests.py

```diff
@@ -33,3 +35,140 @@ def test_clear_cache_celery(self):
         """ Test that the request cache is cleared after a task is run. """
         self._dummy_task.apply(args=(self,)).get()
         self.assertEqual(modulestore().request_cache.data, {})
+
+    def test_request_cached_miss_and_then_hit(self):
+        """
+        Ensure that after a cache miss, we fill the cache and can hit it.
+        """
+        RequestCache.clear_request_cache()
+
+        to_be_wrapped = Mock()
+        to_be_wrapped.return_value = 42
+        self.assertEqual(to_be_wrapped.call_count, 0)
+
+        def mock_wrapper(*args, **kwargs):
+            """Simple wrapper to let us decorate our mock."""
+            return to_be_wrapped(*args, **kwargs)
+
+        wrapped = request_cached(mock_wrapper)
+        result = wrapped()
+        self.assertEqual(result, 42)
+        self.assertEqual(to_be_wrapped.call_count, 1)
+
+        result = wrapped()
+        self.assertEqual(result, 42)
+        self.assertEqual(to_be_wrapped.call_count, 1)
+
+        to_be_wrapped.return_value = 43
```

Comment:
The case of the underlying result changing is covered in the next test, so this last section is redundant.